File: 0107-binary-tree-level-order-traversal-ii/0107-binary-tree-level-order-traversal-ii.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> result;
        queue<TreeNode*> rootque;
        if (root != nullptr) rootque.push(root);
        while (!rootque.empty()){
            int size = rootque.size();
            vector<int> curlev;
            for (int i = 0; i < size; i++){
                TreeNode* cur = rootque.front();
                rootque.pop();
                curlev.push_back(cur->val);
                if (cur->left != nullptr) rootque.push(cur->left);
                if (cur->right != nullptr) rootque.push(cur->right);
            }
            result.push_back(curlev);
        }
        reverse(result.begin(), result.end());
        return result;
    }
};

File: 0107-binary-tree-level-order-traversal-ii/0107-binary-tree-level-order-traversal-ii.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def levelOrderBottom(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        if not root:
            return []
        result  = []
        def traverse(node, level):
            if not node:
                return
            if len(result) == level:
                result.append([])
            traverse(node.left, level+1)
            traverse(node.right, level+1)
            result[level].append(node.val)
        traverse(root, 0)
        
        return reversed(result)